src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Automation/VsBuildManager.cs

```diff
@@ -78,6 +91,14 @@ public object DesignTimeOutputMonikers
         /// </summary>
         public string BuildDesignTimeOutput(string bstrOutputMoniker)
         {
+            _threadingService.VerifyOnUIThread();
```

Comment:
I don't believe this and above is legacy behavior, and its very different to most of the VS APIs that CPS implements. Legacy will never throw off UI-thread, and will always marshal back to the UI thread. CPS almost always lets you call these APIs on a background and will switch to UI thread if it needs to.

Reply:
I guess I misunderstood the "always need to be on the UI thread to call VS services" to mean we could rely on callers to us doing the same, and that since these methods are part of VSLangProj, that applied. I can change this to a UI thread switch inside ExecuteSynchronously.